Integrated player crashes on Android 14

### Describe the bug

1. Go to home.
2. Try to play any video.
3. The player launches and crashes.

Related to [behavior changes in Android 14](https://developer.android.com/about/versions/14/behavior-changes-14#runtime-receivers-exported).

### Logs

```shell
FATAL EXCEPTION: main
Process: org.jellyfin.mobile, PID: 14582
java.lang.SecurityException: org.jellyfin.mobile: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
	at android.os.Parcel.createException(Parcel.java:3041)
	at android.os.Parcel.readException(Parcel.java:3024)
	at android.os.Parcel.readException(Parcel.java:2966)
	at android.app.IActivityManager$Stub$Proxy.registerReceiverWithFeature(IActivityManager.java:5684)
	at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1852)
	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1792)
	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1780)
	at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:755)
	at org.jellyfin.mobile.player.interaction.PlayerNotificationHelper.postNotification(SourceFile:132)
	at org.jellyfin.mobile.player.PlayerViewModel.onPlayerStateChanged(SourceFile:45)
	at o5.a0.invoke(SourceFile:15)
	at v3.a.run(SourceFile:107)
	at i2.e.g(SourceFile:119)
	at o5.i0.J(SourceFile:896)
	at f.r0.run(SourceFile:801)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActivityManagerService.registerReceiverWithFeature(ActivityManagerService.java:13908)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2570)
	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2720)
	at android.os.Binder.execTransactInternal(Binder.java:1339)
	at android.os.Binder.execTransact(Binder.java:1275)
```


### Application version

2.6.0-beta.1

### Where did you install the app from?

Google Play

### Device information

Google Pixel 6a

### Android version

Android 14

### Jellyfin server version

10.8.10

### Which video player implementations does this bug apply to?

- [ ] Web player (default)
- [X] Integrated player (ExoPlayer)
- [ ] External player (VLC, mpv, MX Player)

--- a/app/src/main/java/org/jellyfin/mobile/player/interaction/PlayerNotificationHelper.kt
+++ b/app/src/main/java/org/jellyfin/mobile/player/interaction/PlayerNotificationHelper.kt
@@ -8,6 +8,7 @@ import android.content.Context
 import android.content.Intent
 import android.content.IntentFilter
 import android.graphics.Bitmap
+import androidx.core.content.ContextCompat
 import androidx.core.content.getSystemService
 import androidx.core.graphics.drawable.toBitmap
 import androidx.lifecycle.viewModelScope
@@ -61,7 +62,7 @@ class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinCom
         }
     }
 
-    @Suppress("CyclomaticComplexMethod", "DEPRECATION", "LongMethod")
+    @Suppress("DEPRECATION", "LongMethod")
     fun postNotification() {
         val nm = notificationManager ?: return
         val player = viewModel.playerOrNull ?: return
@@ -123,11 +124,12 @@ class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinCom
             for (notificationAction in PlayerNotificationAction.values()) {
                 filter.addAction(notificationAction.action)
             }
-            if (AndroidVersion.isAtLeastT) {
-                context.registerReceiver(notificationActionReceiver, filter, Context.RECEIVER_NOT_EXPORTED)
-            } else {
-                context.registerReceiver(notificationActionReceiver, filter)
-            }
+            ContextCompat.registerReceiver(
+                context,
+                notificationActionReceiver,
+                filter,
+                ContextCompat.RECEIVER_NOT_EXPORTED,
+            )
         }
     }
 

--- a/app/src/main/java/org/jellyfin/mobile/player/interaction/PlayerNotificationHelper.kt
+++ b/app/src/main/java/org/jellyfin/mobile/player/interaction/PlayerNotificationHelper.kt
@@ -1,6 +1,5 @@
 package org.jellyfin.mobile.player.interaction
 
-import android.app.Application
 import android.app.Notification
 import android.app.NotificationManager
 import android.app.PendingIntent
@@ -38,7 +37,7 @@ import org.koin.core.component.inject
 import java.util.concurrent.atomic.AtomicBoolean
 
 class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinComponent {
-    private val context: Context = viewModel.getApplication<Application>()
+    private val context: Context = viewModel.getApplication()
     private val appPreferences: AppPreferences by inject()
     private val notificationManager: NotificationManager? by lazy { context.getSystemService() }
     private val imageApi: ImageApi = get<ApiClient>().imageApi
@@ -62,7 +61,7 @@ class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinCom
         }
     }
 
-    @Suppress("DEPRECATION")
+    @Suppress("CyclomaticComplexMethod", "DEPRECATION", "LongMethod")
     fun postNotification() {
         val nm = notificationManager ?: return
         val player = viewModel.playerOrNull ?: return
@@ -87,7 +86,8 @@ class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinCom
 
             val notification = Notification.Builder(context).apply {
                 if (AndroidVersion.isAtLeastO) {
-                    setChannelId(Constants.MEDIA_NOTIFICATION_CHANNEL_ID) // Set Notification Channel on Android O and above
+                    // Set notification channel on Android O and above
+                    setChannelId(Constants.MEDIA_NOTIFICATION_CHANNEL_ID)
                     setColorized(true)
                 } else {
                     setPriority(Notification.PRIORITY_LOW)

--- a/app/src/main/java/org/jellyfin/mobile/player/interaction/PlayerNotificationHelper.kt
+++ b/app/src/main/java/org/jellyfin/mobile/player/interaction/PlayerNotificationHelper.kt
@@ -123,7 +123,11 @@ class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinCom
             for (notificationAction in PlayerNotificationAction.values()) {
                 filter.addAction(notificationAction.action)
             }
-            context.registerReceiver(notificationActionReceiver, filter)
+            if (AndroidVersion.isAtLeastT) {
+                context.registerReceiver(notificationActionReceiver, filter, Context.RECEIVER_NOT_EXPORTED)
+            } else {
+                context.registerReceiver(notificationActionReceiver, filter)
+            }
         }
     }
 

